MREUnityRuntime/MREUnityRuntimeLib/Core/Actor.cs

```diff
@@ -1235,32 +1235,53 @@ private void OnSetAnimationState(SetAnimationState payload, Action onCompleteCal
             onCompleteCallback?.Invoke();
         }
 
-        [CommandHandler(typeof(SetSoundState))]
-        private void OnSetSoundState(SetSoundState payload, Action onCompleteCallback)
+        [CommandHandler(typeof(SetMediaState))]
+        private void OnSetMediaState(SetMediaState payload, Action onCompleteCallback)
         {
-            if (payload.SoundCommand == SoundCommand.Start)
+            if (payload.MediaCommand == MediaCommand.Start)
             {
+                if (_mediaInstances == null)
+                {
+                    _mediaInstances = new Dictionary<Guid, System.Object>();
+                }
                 AudioSource soundInstance = App.SoundManager.TryAddSoundInstance(this, payload.Id, payload.SoundAssetId, payload.Options, payload.StartTimeOffset);
                 if (soundInstance)
                 {
-                    if (_soundInstances == null)
+                    _mediaInstances.Add(payload.Id, soundInstance);
+                }
+                else
+                {
+                    var factory = MREAPI.AppsAPI.VideoPlayerFactory
+                        ?? throw new ArgumentException("Cannot movie player not implemented library.");
```

Comment:
The semiotics of this error message are beyond me.

Reply:
yeah, that was weird. 

---
In reply to: [302633494](https://github.com/Microsoft/mixed-reality-extension-unity/pull/86#discussion_r302633494) [](ancestors = 302633494)